### PR TITLE
Clean-up, verification, tests

### PR DESCRIPTION
**Pull request by kupermind**
- Cleaning the repository;
- Adding more verifications;
- Adding tests.

### CODE REVIEW

File: solidity/library/spl_token.sol [hunk 1]
```diff
@@ -60,66 +60,17 @@ library SplToken {
 			AccountMeta({pubkey: authority, is_writable: false, is_signer: true})
 		];
 
-		tokenProgramId.call{accounts: metas}(instr);
-	}
-
-	/// Mint new tokens. The transaction should be signed by the mint authority keypair
```
Comment (kupermind) [inline solidity/library/spl_token.sol:66]: Cleaning unused functions.

File: solidity/liquidity_lockbox.sol [hunk 2]
```diff
@@ -116,9 +123,25 @@ contract nft_token {
...
         Position positionData = _getPositionData(tx.accounts.position, tx.accounts.positionMint.key);
 
-        // TODO: Do the liquidity check for max(uint64) value as it is provided as uint128 from the LP provider
         uint64 positionLiquidity = uint64(positionData.liquidity);
 
+        // Check that the mint of the user position ATA matches the position mint
+        address positionMint = tx.accounts.userPositionAccount.data.readAddress(0);
+        if (positionMint != tx.accounts.positionMint.key) {
+            revert("Wrong user position ATA");
+        }
```
Comment (kupermind) [inline solidity/liquidity_lockbox.sol:132]: Additional verifications against other possible accounts.

File: solidity/liquidity_lockbox.sol [hunk 3]
```diff
@@ -144,39 +167,6 @@ contract nft_token {
...
-    @mutableAccount(position)
-    @account(positionTokenAccount)
-    @mutableAccount(tokenOwnerAccountA)
-    @mutableAccount(tokenOwnerAccountB)
-    @mutableAccount(tokenVaultA)
-    @mutableAccount(tokenVaultB)
-    @mutableAccount(tickArrayLower)
-    @mutableAccount(tickArrayUpper)
-    // Transfer with PDA
-    function decreaseLiquidity(uint128 amount, uint64 minA, uint64 minB) external {
```
Comment (kupermind) [inline solidity/liquidity_lockbox.sol:159]: Test function, cleaning up

File: solidity/liquidity_lockbox.sol [hunk 4]
```diff
@@ -308,9 +298,11 @@ contract nft_token {
         return _getPositionData(tx.accounts.position, tx.accounts.positionMint.key);
     }
 
-    function getLiquidityAmountsAndPositions(uint64 amount) external view returns (uint64[], address[], address[]) {
+    function getLiquidityAmountsAndPositions(uint64 amount)
+        external view returns (uint64[] positionAmounts, address[] positionAddresses, address[]positionPdaAtas)
+    {
         uint64 totalLiquidity = 0;
-        uint64 numPositions = 0;
+        uint32 numPositions = 0;
```
Comment (kupermind) [inline solidity/liquidity_lockbox.sol:305]: Returning actual values now, also the array size can only be set in `uint32`

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**